### ISSUE TITLE
Don't use "postgres" superuser for running tests

### ISSUE THREAD

**Issue by jstolarek** [links-lang/links#66]
I just spent about an hour trying to figure out why [the following line from the `run-tests` script](https://github.com/links-lang/links/blob/5921202b2b5e1926e146032fa5d3eaeb9db88111/run-tests#L24) does not work on my machine:

```
psql -v ON_ERROR_STOP=1 -q -U postgres -d links -f $s
```
It turns out that Debian (and supposedly Ubuntu also) blocks access to `postgres` superuser via TCP. The workaround that works on Debian is to explicitly connect to localhost using a *nix socket since the rules are less strict then:

```
psql -v ON_ERROR_STOP=1 -q -U postgres -d links -h localhost -f $s
```

I believe the proper solution to this would be to stop using the superuser for operations on the database. Perhaps we could assume that a user that runs the test script has a `links` database that she has access to? Then we could just say:

```
psql -v ON_ERROR_STOP=1 -q -d links -f $s
```

What others think?

**Comment by fehrenbach** [comment]
This is strange, the man page says:
> If you omit the host name, psql will connect via a Unix-domain socket to a server on the local host, or via TCP/IP to localhost on machines that don't have Unix-domain sockets.

Anyway, it is the way it is because it works on my machine and it works on Travis (Ubuntu trusty) without any further setup.

Current user would be nice, because you can just leave out the user for psql :)
On the other hand, the user appears in the Links config file, see https://github.com/links-lang/links/blob/sessions/tests/database/config#L2
I'm not sure that's actually needed, or what happens if you leave out the user.

**Comment by jstolarek** [comment]
> On the other hand, the user appears in the Links config file, see https://github.com/links-lang/links/blob/sessions/tests/database/config#L2
I'm not sure that's actually needed, or what happens if you leave out the user.

I believe we would need to be smarter in the PostgreSQL driver. Currently it just assumes that there is a user name:

```ocaml
let get_pg_database_by_string args =
  match Utility.split_string args ':' with
    | (name::host::port::user::pass::_) ->
        (new pg_database host port name user pass,
         Value.reconstruct_db_string (driver_name, args))
    | _ ->
        failwith "Insufficient arguments when establishing postgresql connection"
```
I'll try to figure out how to acquire current user name.

**Comment by jstolarek** [comment]
Done. It is no OK to write database arguments like this:

```
database_args=localhost:5432::links
```

If the user field is left empty it will default to the name of the user running the process.

I'll test whether dropping not using postgres as database user breaks Travis. Stefan, can you test on your machine as well?

**Comment by jstolarek** [comment]
Yup, Travis looks happy. @fehrenbach, let me know whether this works on your machine as well.

**Comment by fehrenbach** [comment]
`js-no-postgres-user` seems to work for me.

I saw that you added install documentation on the wiki. This is slightly different from the instructions in INSTALL. Also, I think we should recommend using ocamlfind to find POSTGRESQL_LIBDIR, like the Makefile.sample.config does. Maybe our next buildsystem could even do that automatically?

**Comment by jstolarek** [comment]
> `js-no-postgres-user` seems to work for me.

Merged into `sessions`. Others, please yell if this change causes problems on your system.

> This is slightly different from the instructions in INSTALL. Also, I think we should recommend using ocamlfind to find POSTGRESQL_LIBDIR, like the Makefile.sample.config does.

Good point! I realized that `INSTALL` exists only after I have figured out everything by myself and missed the fact that there is a better way of finding `postgresql` installation. I'll update the wiki.

> Maybe our next buildsystem could even do that automatically?

I'll take this discussion to #62.
